### DISCUSSION TITLE
fp-like isPast, isToday, etc implementation

### DISCUSSION THREAD

**Discussion by willisplummer** [js-fns/date-fns#2321] [Ideas] [answered]
I noticed that the fp library doesn't expose any functions that use `Date.now()` internally.. What do you think about adding versions of these fn's that allows for injection of `now` (in general, this makes testing so much easier)

e.g.

```
export const isPast = (date: Date, now: Date = new Date()): boolean => (date.getTime() - now.getTime) < 0
```

consider the fn `foo`:

```
import { isPast } from 'date-fns';

export const foo = (date: Date): string => isPast(date) ? 'past' : 'future';
```

currently, we have to mock out `isPast` in order to test foo (or use arbitrarily futuristic and ancient dates)

if we have the ability to inject `now` into the fn for testing, we can simply update foo to also take an optional `now`:

```
export const foo = (date: Date, now: Date = new Date()): string => isPast(date, now) ? 'past' : 'future';
```

and then hardcode the current time in our test

```
expect(foo(new Date(100), newDate(0))).toBe('future')
expect(foo(new Date(100), newDate(200))).toBe('past')
```

i find this is a really great pattern for working around the concept of now with date functions and would love to see it supported by the lib, but i get that it represents a bit of a paradigm shift.. any way figured i'd suggest it. eager to hear your thoughts!

**Comment by kossnocorp** [comment, marked as answer]
Hey! We have such a function. It's `isSameDay` 😉 I think using `isPast(now, date)` in FP code is just confusing. Unless you have better reasons, I would like to keep it as is and recommend use `isSameDay` (and other FP-friendly alternatives).
